site/content/blog/2020-04-08-automate-security-testing-with-zap-and-github-actions/index.md

```diff
@@ -0,0 +1,81 @@
+a web application without doing any active attacks. 
+
+Furthermore, GitHub Actions makes it easier to automate how to scan and secure web applications at scale. Actions let you 
+write scripts that are triggered based on certain events in your GitHub repo such as — creating a new issue, pushing a 
+commit, or on a scheduled basis. This makes it convenient and easy to automate security testing and to run ZAP scans 
+without involving any of your own infrastructures. The ZAP baseline action is available in the Github Marketplace under 
+the actions/security category.
+
+The [ZAP baseline-action](https://github.com/marketplace/actions/owasp-zap-baseline-scan) can be configured to periodically 
+scan a publicly available web application. Also, ZAP baseline-action can be configured to public and private repositories as well.
+Based on the scan results ZAP will maintain an active issue in GitHub repository. The action will update the issue if it identifies 
+any new or resolved alerts and will close the issue if all the alerts have been resolved. A detailed report is attached with 
+the issue to get more information regarding the identified alerts. The report is available in HTML and markdown formats.
+
+The following image shows how the GitHub Actions will notify the user with consequent scans.
+
+![zap-issue](./images/zap-issue.png)
+
+The ZAP baseline action scan runs the ZAP spider against the specified web application for (by default) 1 minute and 
+then waits for the passive scanning to complete before reporting the results. This means that the script doesn't 
+perform any actual ‘attacks’ and will run for a relatively short period of time (a few minutes at most).   
+
+The following YAML config shows how to configure the action to scan a publicly accessible web application. 
+In order to start scanning, create a file called `zap_scan.yaml` inside the `.github/workflows` 
+folder in your repository. 
+
+```yaml
+on:
+  schedule:
+    # Runs every day At 01:00.
+    - cron:  '0 1 * * *'
+
+jobs:
+  zap_scan:
+    runs-on: ubuntu-latest
+    name: Scan ZAP website
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: master
+      - name: ZAP Scan
+        uses: zaproxy/action-baseline@v0.1.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          target: 'https://www.zaproxy.org/'
+```
+
+The ZAP baseline action can be also used to scan a locally built web application. The general idea is to build and serve your web 
+application and make it available via a headless browser so ZAP can access the site via the localhost.
+
+By default, baseline scan reports all alerts as WARNings but you can specify a config file that can change any rules to FAIL or IGNORE.
+The following shows a sample rules file. Create the `rules.tsv` file inside your repository and make sure to update the action 
```

Comment:
Add example with the field?

Reply:
Do you mean the YAML config to run the action?

Reply:
Just the field, e.g.:
```yaml
          rules_file_name: '.github/workflows/rules.tsv'
```

Reply:
Okay, I have refactored the first yaml to include very minimal configs and in the second part, I have added these fields.